ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/console/DefaultOutputCustomizer.java

```diff
@@ -0,0 +1,234 @@
+import org.eclipse.che.ide.api.resources.Container;
+import org.eclipse.che.ide.api.resources.File;
+import org.eclipse.che.ide.api.resources.Resource;
+import org.eclipse.che.ide.resource.Path;
+
+import com.google.gwt.regexp.shared.MatchResult;
+import com.google.gwt.regexp.shared.RegExp;
+import com.google.gwt.user.client.Timer;
+import com.google.inject.Inject;
+
+/**
+ * Default customizer adds an anchor link to the lines that match a stack trace
+ * line pattern and installs a handler function for the link. The handler parses
+ * the stack trace line, searches for the candidate Java files to navigate to,
+ * opens the first file (of the found candidates) in editor and reveals it to
+ * the required line according to the stack trace line information
+ */
+public class DefaultOutputCustomizer implements OutputCustomizer {
+
+    private static final RegExp LINE_AT = compile("(\\s+at .+)");
+    private static final RegExp LINE_AT_EXCEPTION = compile("(\\s+at address:.+)");
+
+    private AppContext appContext;
+    private EditorAgent editorAgent;
+
+    @Inject
+    public DefaultOutputCustomizer(AppContext appContext, EditorAgent editorAgent) {
+        this.appContext = appContext;
+        this.editorAgent = editorAgent;
+
+        exportAnchorClickHandlerFunction();
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.eclipse.che.ide.extension.machine.client.outputspanel.console.
+     * OutputCustomizer#canCustomize(java.lang.String)
+     */
+    @Override
+    public boolean canCustomize(String text) {
+        return (LINE_AT.exec(text) != null && LINE_AT_EXCEPTION.exec(text) == null);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see org.eclipse.che.ide.extension.machine.client.outputspanel.console.
+     * OutputCustomizer#customize(java.lang.String)
+     */
+    @Override
+    public String customize(String text) {
+        String customText = text;
+
+        MatchResult matcher = LINE_AT.exec(text);
+        if (matcher != null) {
+            try {
+                int start = text.indexOf("at", 0) + "at".length(), openBracket = text.indexOf("(", start),
+                        column = text.indexOf(":", openBracket), closingBracket = text.indexOf(")", column);
+                String qualifiedName = text.substring(start, openBracket).trim();
+                String fileName = text.substring(openBracket + "(".length(), column).trim();
+                int lineNumber = Integer.valueOf(text.substring(column + ":".length(), closingBracket).trim());
+                customText = text.substring(0, openBracket + "(".length());
+                customText += "<a href='javascript:open(\"" + qualifiedName + "\", \"" + fileName + "\", " + lineNumber
+                        + ");'>";
+                customText += text.substring(openBracket + "(".length(), closingBracket);
+                customText += "</a>";
+                customText += text.substring(closingBracket);
+                text = customText;
```

Comment:
String.format/MessageFormat.format/Formatter.format are not available for GWT application and it doesn't look like a GWT replacement exists.